Source/Core/VideoCommon/VertexShaderManager.cpp

```diff
@@ -690,6 +690,24 @@ void VertexShaderManager::ResetView()
 	bProjectionChanged = true;
 }
 
+void VertexShaderManager::TransformToClipSpace(const float* data, float *out)
+{
+	const float *world_matrix = (const float *)xfmem.posMatrices + g_main_cp_state.matrix_index_a.PosNormalMtxIdx * 4;
```

Comment:
if a game uses per vertex matrix for a reference triangle you will get wrong results here, maybe is beter to check the games to see if any of them is using this functionality.The implementation is easy is just look at the last 4 bytes of the vertex as a u32 and index the matrix xfmem area  to get the correct matrix.